src/api/spec/controllers/webui/package_controller_spec.rb

```diff
@@ -397,4 +397,47 @@
       end
     end
   end
+
+  describe "DELETE #remove_file" do
+    before do
+      login(user)
+      Package.any_instance.stubs(:delete_file).returns(true)
+    end
+
+    def remove_file_post
+      post :remove_file, project: user.home_project, package: source_package, filename: 'the_file'
+    end
+
+    context "with successful backend call" do
+      before do
+        remove_file_post
```

Comment:
tbh, I don't think the duplication of the post request would be too bad. Especially in a test where readability is often more important than reducing duplication.

But I am fine with both ways:)